### DISCUSSION TITLE
Filename sanitization diference on small filenames

### DISCUSSION THREAD

**Discussion by oscar-costa** [pocketbase/pocketbase#1783] [Q&A] [answered]
Hi.
1. When uploading a file i'm getting diferente file sanitazed filenames depending on the total characters size of the file:
".png"  saves file as "vL1TrTalta_80YtGLKxce.png
"a.png"  saves file as "aARklLaC8vl_UJ8KM327zG.png"
"aa.png"  saves file as "aaEAp59F6oRQ_PgRWWEEwJO.png"
"aaa.png"  saves file as "aaa_3TxAs5H7cR.png"

If the filename as less than 3 characters long it adds adicional characters to the sanitization, plus the expected suffix with a random 10 chars.

2. It would be great if we had the posibility to access/download files with their original names, through the SDK. For projects with some document management features it could make all sense.

**Comment by ganigeorgiev** [comment, marked as answer]
This is intentional since currently all files are public (as long as you know their url) and files with filenames less than 3 characters are concatenated additionally with a random part to make them long enough. Eventually once we introduce private/protected files (most likely in v0.14/v0.15) I'll consider limiting the random parts.

There are no plans for now to allow downloading files with their original names (_your can also check https://github.com/pocketbase/pocketbase/discussions/210_). If you want to show users the original name, you can store it as a collection field alongside the file before upload.

**Comment by oscar-costa** [comment]
Ok thanks
